content/docs/observable.md

```diff
@@ -0,0 +1,202 @@
+# Observable
+
+> [tip box yellow] This functionality was added in v2.0.0-alpha.202 and is still experimental.
+
+### TLDR
+Imba provides MobX-like reactivity implemented as decorators.
+- Mark class and tag members as observable with the `@observable` decorator.
+- Methods decorated with `@autorun` will be reactively invoked whenever
+referenced observables change.
+- Getters decorated with `@computed` will be cached on first use
+and only recomputed whenever referenced observables change.
+
+See also [additional details](#additional-details).
+
+### Guide
+In Imba, it's important to keep in mind that [commit](/api/imba/commit) is called after every handled event.
+This means that state changes which occur inside of event handlers will be reflected in the next render.
+We can apply this knowledge to implement a simplistic search:
+#### The Inefficient Approach
+```imba
+# [preview=lg]
+css body d:flex fld:column jc:start ai:center p:5
+css input bg:blue0 mb:1 rd:2 outline:none p:1 3 bd:1px solid blue3
+	caret-color:blue4 bxs:sm bg@focus:blue1 transition:background 300ms
+css div bg:gray1 rd:2 p:1 3 bdb:1px solid gray2 c:gray6 mb:1 bxs:xs
+# ---
+let query = ""
+let words = ["hello", "world", "hi"]
```

Comment:
Let's use a nicer set of words, with some theme, like "apple", "orange", "strawberry"